backend/app/process_pdfs.py

```diff
@@ -0,0 +1,78 @@
+import os
+import camelot
+import fnmatch
+import pandas as pd
+
+#TO-DO: Check for PDFs resulting in empty table, edit tables to ensure data in formed correctly in CSV file
+# Idea: check for cells with \n characters using the tables' .data attribute and separate them into rows 
+# if each column has the same number of elements split by \n
+
+#Search through data folder & only collect PDF files
+def find_files(directory, pattern):
+    matches = []
+    for root, _, filenames in os.walk(directory):
+        for filename in fnmatch.filter(filenames, pattern):
+            matches.append(os.path.join(root, filename))
+    return matches  
+
+directory = 'data'
+pattern = '*.pdf'
+found_files = find_files(directory, pattern)
+print(found_files)
+
+
+for file in found_files:
+    # For each of the datasets, ensure that we are only converting pages with data 
+    if file == 'data/BJP2022-2023.pdf':
```

Comment:
Generally, you should avoid using raw strings when working with file paths, as you do here. In an OS like Windows, file paths typically use '\' instead of '/', so running this script results in the error "page_numbers" not found since the equality statements use '/'. If you ever need to compare paths like this, you should use something like Path from pathlib instead to ensure OS cross-compatibility:
```
>>> from pathlib import Path
>>> Path("p/o")
WindowsPath('p/o')
>>> Path("p/o") == Path(r"p\o")
True
```
However, this comparison can be avoided if you implement the suggestion in my next comment (lines 24 - 63).